nrf_security: Enable PSA Key types in samples and configurations (#11811)
nrfconnect/sdk-nrf: doc/nrf/libraries/nrf_security/doc/driver_config.rst

Enabling or disabling PSA driver specific configurations controls the support for a given algorithm, per driver.

Key type configuration
**********************

> To enable block or stream cipher algorithms, set one or more of the following Kconfig options:

Reviewer: AFAIK RSA does not use a block or stream cipher.

Perhaps we should describe this in a different way:

Algorithms use keys with some flexibility in which keys can be used for which operations.
But to use a PSA key it's corresponding Kconfig option must first be enabled.

The Kconfig options follow a specific naming pattern: the PSA key PSA_KEY_TYPE_<type> is enabled by the Kconfig option CONFIG_PSA_WANT_KEY_TYPE_<type>.

For information about which keys can be used with specific algorithms, please refer to the PSA Crypto API documentation

> To enable block or stream cipher algorithms, set one or more of the following Kconfig options:

Reviewer 2: Shall we just say "cryptographic algorithms", as then it includes asymmetric crypto, and the key agreement algorithms as well
```suggestion
To enable cryptographic algorithms, set one or more of the following Kconfig options:
```

+-----------------------+-------------------------------------------------------------+
| Key type              | Configuration option                                        |
+=======================+=============================================================+
| AES                   | :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_AES`              |
+-----------------------+-------------------------------------------------------------+
| Chacha20              | :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_CHACHA20`         |
+-----------------------+-------------------------------------------------------------+

> | ECC Key Pair          | :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY`   |

Reviewer: ```suggestion
| ECC Key Pair          | :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_ECC_KEY_PAIR`   |
```

Author: Thanks.

+-----------------------+-------------------------------------------------------------+
| ECC Public Key        | :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY`   |
+-----------------------+-------------------------------------------------------------+
| RSA Key Pair          | :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_RSA_KEY_PAIR`     |
+-----------------------+-------------------------------------------------------------+
| RSA Public Key        | :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY`   |
+-----------------------+-------------------------------------------------------------+
| DH Key Pair           | :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_DH_KEY_PAIR`      |
+-----------------------+-------------------------------------------------------------+
| DH Public key         | :kconfig:option:`CONFIG_PSA_WANT_KEY_TYPE_DH_PUBLIC_KEY`    |
+-----------------------+-------------------------------------------------------------+

Key type support
================

Key type support for each driver:

> +-----------------------+---------------------------+----------------------------+

Reviewer 3: This table doesn't render properly (additional + sign needed on lines 67, 69, 71 (...) to indicate a separate cell)
![image](https://github.com/nrfconnect/sdk-nrf/assets/29537761/5b002f30-4613-4353-83d9-cd2a4e147c76)

Enabling or disabling PSA driver specific configurations controls the support for a given algorithm, per driver.

Key type configuration
**********************

> To enable block or stream cipher algorithms, set ore or more of the following Kconfig options:

Reviewer: ```suggestion
To enable block or stream cipher algorithms, set one or more of the following Kconfig options:
```

+-----------------------+---------------------------------------------------+
| Hash algorithm        | Configuration option                              |
+=======================+===================================================+

> | SHA-1 (unsafe)        | :kconfig:option:`CONFIG_PSA_WANT_ALG_SHA_1`       |

Reviewer: ```suggestion
| SHA-1 (weak)        | :kconfig:option:`CONFIG_PSA_WANT_ALG_SHA_1`       |
```

same elsewhere. See Georgios's review.

Reviewer 2: Agree the PSA Crypto API doc also uses the word weak for it I think it would be nice to be aligned with them, and we use weak in L460/461
https://armmbed.github.io/mbed-crypto/html/api/ops/hashes.html

+-----------------------+---------------------------------------------------+
| SHA-512               | :kconfig:option:`CONFIG_PSA_WANT_ALG_SHA_512`     |
+-----------------------+---------------------------------------------------+

> | MD5 (unsafe)          | :kconfig:option:`CONFIG_PSA_WANT_ALG_MD5`         |

Reviewer 2: ```suggestion
| MD5 (weak)          | :kconfig:option:`CONFIG_PSA_WANT_ALG_MD5`         |
```

+-----------------------+---------------------------------------------------------------+---------------------------------------------------------------+
| Hash algorithm        |  nrf_cc3xx driver support                                     | nrf_oberon driver support                                     |
+=======================+===============================================================+===============================================================+

> | SHA-1 (unsafe)        | :kconfig:option:`CONFIG_PSA_CRYPTO_DRIVER_ALG_SHA_1_CC3XX`    | :kconfig:option:`CONFIG_PSA_CRYPTO_DRIVER_ALG_SHA_1_OBERON`   |

Reviewer 2: ```suggestion
| SHA-1 (weak)        | :kconfig:option:`CONFIG_PSA_CRYPTO_DRIVER_ALG_SHA_1_CC3XX`    | :kconfig:option:`CONFIG_PSA_CRYPTO_DRIVER_ALG_SHA_1_OBERON`   |
```

+-----------------------+---------------------------------------------------------------+---------------------------------------------------------------+
| SHA-512               | Not supported                                                 | :kconfig:option:`CONFIG_PSA_CRYPTO_DRIVER_ALG_SHA_512_OBERON` |
+-----------------------+---------------------------------------------------------------+---------------------------------------------------------------+

> | MD5 (unsafe)          | Not supported                                                 | Not supported                                                 |

Reviewer 2: ```suggestion
| MD5 (weak)          | Not supported                                                 | Not supported                                                 |
```
